### PR TITLE
update the backwardsCompatibility to docs

### LINKED ISSUE #81675: AppBarTheme titleTextStyle is not work
```dart
var appTheme = ThemeData.from(colorScheme: ColorScheme.light()).copyWith(
  appBarTheme: AppBarTheme(
    titleTextStyle: TextStyle(
      fontSize: 18,
      fontWeight: FontWeight.w600,
      color: AppColors.primaryText,
    ),
  ),
// ...
);
```

```dart
class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return ScreenUtilInit(
        designSize: Size(375, 667),
        builder: () => GetMaterialApp(
              theme: appTheme,
              initialRoute: AppRoutes.index,
              getPages: AppPages.pages,
            ));
  }
}
```

```dart
return Scaffold(
      appBar: AppBar(
       // but the style set by titleTextStyle is now work!!!
        title: Text(
          'test',
        ),
      ),
      body: _buildBody(),
    );
```

```
Flutter 2.0.6 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 1d9032c7e1 (3 days ago) • 2021-04-29 17:37:58 -0700
Engine • revision 05e680e202
Tools • Dart 2.12.3
```

### PR DESCRIPTION
**Pull request by Abhishek01039**
Fixes https://github.com/flutter/flutter/issues/81675

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Abhishek01039** [comment]
Hello @HansMuller 
can you please review this PR?
Thanks

File: packages/flutter/lib/src/material/app_bar_theme.dart [hunk 1]
```diff
@@ -166,6 +169,9 @@ class AppBarTheme with Diagnosticable {
   /// Overrides the default value of [AppBar.titleTextStyle]
   /// property in all descendant [AppBar] widgets.
   ///
+  /// Please set [backwardsCompatibility] true if you want to override
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/app_bar_theme.dart:172]: I think you want to say that if this property is specified, then backwardsCompatibility should be true. Although one could override the `get titleTextStyle` method, that doesn't seem like the case that's worth documenting.

File: packages/flutter/lib/src/material/app_bar_theme.dart [hunk 2]
```diff
@@ -157,6 +157,9 @@ class AppBarTheme with Diagnosticable {
   /// Overrides the default value for the obsolete [AppBar.toolbarTextStyle]
   /// property in all descendant [AppBar] widgets.
   ///
+  /// Please set [backwardsCompatibility] true if you want to override
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/app_bar_theme.dart:160]: Same comment as below

File: packages/flutter/lib/src/material/app_bar_theme.dart (hunk 1 above)
Comment (Abhishek01039) [reply packages/flutter/lib/src/material/app_bar_theme.dart]: Thanks @HansMuller

File: packages/flutter/lib/src/material/app_bar_theme.dart [hunk 3]
```diff
@@ -157,8 +157,7 @@ class AppBarTheme with Diagnosticable {
   /// Overrides the default value for the obsolete [AppBar.toolbarTextStyle]
   /// property in all descendant [AppBar] widgets.
   ///
-  /// Please set [backwardsCompatibility] true if you want to override
-  /// this method.
+  /// if this property is specified, then [backwardsCompatibility] should be true.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/app_bar_theme.dart:160]: Please start sentences with a capital letter

Comment (Abhishek01039) [reply packages/flutter/lib/src/material/app_bar_theme.dart]: Sure thanks

**Review by HansMuller — APPROVED** [review]
LGTM. Will land this after Ian gives it a no-tests exemption.

**Comment by Hixie** [comment]
test-exempt: only affects comments

**Pull request merged**